cg/meta/workflow/nf_analysis.py

```diff
@@ -737,3 +739,34 @@ def get_cases_to_store(self) -> list[Case]:
             if self.trailblazer_api.is_latest_analysis_completed(case_id=case.internal_id)
             or self.trailblazer_api.is_latest_analysis_qc(case_id=case.internal_id)
         ]
+
+    def clean_workflow_run_dir(self, yes: bool, case_id: str, dry_run: bool = False):
+        """Remove workflow run directory."""
+
+        self.check_analysis_ongoing(case_id=case_id)
+        analysis_path: list[Path] | Path = self.get_case_path(case_id)
+
+        if dry_run:
+            LOG.info(f"Would have deleted: {analysis_path}")
+            return EXIT_SUCCESS
+
+        self.clean_run_dir(case_id=case_id, yes=yes, case_path=analysis_path)
+
+    def clean_past_run_dirs(self, before_str: str, yes: bool = False, dry_run: bool = False):
+        """Clean up of old case run dirs."""
+
+        before = parse_date(before_str)
+        possible_cleanups = self.get_analyses_to_clean(before=before)
+        LOG.info(f"Cleaning {len(possible_cleanups)} analyses created before {before}")
+
+        for analysis in possible_cleanups:
+            case_id = analysis.case.internal_id
+            try:
```

Comment:
check the type hints
```suggestion
        before: datetime = parse_date(before_str)
        possible_cleanups: list[Analysis] = self.get_analyses_to_clean(before=before)
        LOG.info(f"Cleaning {len(possible_cleanups)} analyses created before {before}")

        for analysis in possible_cleanups:
            case_id:str  = analysis.case.internal_id
            try:
```